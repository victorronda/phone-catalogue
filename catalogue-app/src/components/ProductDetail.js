import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import ClipLoader from 'react-spinners/ClipLoader';

const ProductDetail = ({ phones, id }) => {

	const phone = phones.filter(phone=>phone.id === Number(id))[0];

	return (
		<div>
			{phone ? (
				<div>
                    <img src={phone.imageFileName} alt={phone.name}/>
					<h1>{phone.name}</h1>
				</div>
			) : (
				<ClipLoader />
			)}
		</div>
	);
};

const mapStateToProps = (state) => {
	return {
		phones: state.requestAllProducts.phones
	};
};

export default connect(mapStateToProps, {})(ProductDetail);
