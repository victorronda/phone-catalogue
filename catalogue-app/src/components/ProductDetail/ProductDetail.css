.img-detail {
    width: 100%;
    height: 400vh;
    object-fit: cover;
}