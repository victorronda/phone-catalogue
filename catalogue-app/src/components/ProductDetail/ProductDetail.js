import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import ClipLoader from 'react-spinners/ClipLoader';

const ProductDetail = ({ phones, id }) => {
	const phone = phones.filter((phone) => phone.id === Number(id))[0];

	return (
		<div>
		{phone ? (
			<div className="container d-flex justify-content-center align-center rounded-lg my-3" >
				<div className="row text-center shadow-lg" >
					<div className="col-lg-4 col-md-6 col-sm-8 bg-light rounded-lg">
						<img className="img-fluid" src={phone.imageFileName} alt={phone.name} width="auto"
                    height="auto"/>
					</div>
					<div className="card col-lg-8 col-md-6 col-sm-8 pt-3">
						<h1 className="mb-3 mt-3">{phone.name}</h1>
						<h3 className="mb-4 mt-2">{phone.price}â‚¬</h3>
						<h5>Technical specifications</h5>
						<p className="mt-2"> 
							<b>Manufacturer: </b> {phone.manufacturer}
						</p>
						<p className="">
							<b>Color: </b> {phone.color}
						</p>
						<p className="">
							<b>Screen: </b> {phone.screen}
						</p>
						<p className="">
							<b>Processor: </b> {phone.processor}
						</p>
						<hr/>
						<h5>Description</h5>
						<p className="mt-2">{phone.description}</p>
						<Link className="btn btn-info w-25 mx-auto my-3" to="/">Go back</Link>
					</div>
				</div>
				</div>
			) : (
				<div >
					<ClipLoader color={"#123abc"}size={250} />
				</div>
			)}
		</div>
	);
};

const mapStateToProps = (state) => {
	return {
		phones: state.requestAllProducts.phones
	};
};

export default connect(mapStateToProps, {})(ProductDetail);
