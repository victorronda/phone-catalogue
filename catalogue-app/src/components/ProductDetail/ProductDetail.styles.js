import styled from 'styled-components';

export const FlexContainerDetail = styled.div`
    background: rgba(255,255,255,0.3);
    max-width: 60vw;
    margin: 0 auto;
    display: flex;
    flex-direction: row;
    color: white;
`;

export const ImageDetail = styled.img`
    min-width: 20vw;
    max-width: 20vw;
`;

