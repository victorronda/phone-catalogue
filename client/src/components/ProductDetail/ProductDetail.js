import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';


const ProductDetail = ({ phones, id }) => {
	const phone = phones.filter((phone) => phone.id === Number(id))[0];

	return (
		<div>
			{phone ? (
				<div className="container d-flex justify-content-center align-center rounded-lg my-3">
					<div className="row text-center shadow-lg">
						<div className="col-lg-4 col-md-6 col-sm-8 bg-light rounded-lg">
							<img
								className="img-fluid"
								src={phone.imageFileName}
								alt={phone.name}
								width="auto"
								height="300"
							/>
						</div>
						<div className="card col-lg-8 col-md-6 col-sm-8 pt-3">
							<h1 className="mb-3 mt-3">{phone.name}</h1>
							<h3 className="mb-4 mt-2">{phone.price}â‚¬</h3>
							<h5>Technical specifications</h5>
							<p className="mt-2">
								<b>Manufacturer: </b> {phone.manufacturer}
							</p>
							<p>
								<b>Color: </b> {phone.color}
							</p>
							<p>
								<b>Screen: </b> {phone.screen}
							</p>
							<p>
								<b>Processor: </b> {phone.processor}
							</p>
							<hr />
							<h5>Description</h5>
							<p className="mt-2">{phone.description}</p>
							<Link className="btn btn-info w-25 mx-auto my-3" to="/">
								Go back
							</Link>
						</div>
					</div>
				</div>
			) : (
				<div class="d-flex justify-content-center">
					<div class="spinner-border text-danger" style={{ width: '5rem', height: '5rem' }} role="status">
						<span class="sr-only">Loading...</span>
					</div>
				</div>
			)}
		</div>
	);
};

const mapStateToProps = (state) => {
	return {
		phones: state.requestAllProducts.phones
	};
};

export default connect(mapStateToProps, {})(ProductDetail);
