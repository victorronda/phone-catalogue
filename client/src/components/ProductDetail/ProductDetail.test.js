import { shallow } from 'enzyme';
import React from 'react';
import ProductDetail from './ProductDetail';

it('should render ProductDetail component', () => {
    let wrapper;
    beforeEach(() => {
        const mockPhones = { phones };
        const mockURLId = 1
        wrapper = shallow(<ProductDetail id={mockURLId} phones={mockPhones} />)
    })
    expect(wrapper).toMatchSnapshot();
});