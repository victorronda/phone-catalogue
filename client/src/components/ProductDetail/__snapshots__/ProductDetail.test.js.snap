// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProductDetail component 1`] = `undefined`;
